controllers/nodefeaturediscovery_controller.go

```diff
@@ -136,11 +175,15 @@ func (r *NodeFeatureDiscoveryReconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{Requeue: true}, err
 	}
 
-	// apply components
+	// Apply NFD components upon initializing NFD
 	r.Log.Info("Ready to apply components")
-
 	nfd.init(r, instance)
 
+	// For each resource function stored in "nfd.controls", it is
+	// necessary to determine if the relevant resource is Ready or
+	// NotReady. This loop stops when either a resource is returned
+	// as NotReady or when the loop has reached the last element of
+	// the nfd.controls list -- determined by nfd.last()
```

Comment:
I think the code below is very self-explaining. I'd just be less verbose. One suggestion:

```
// Run through all control functions, return an error on any NotReady resource.
```